tests/protractor/e2e/registration-by-sms.js

```diff
@@ -0,0 +1,279 @@
+  const refresh = () => {
+    browser.driver.navigate().refresh();
+    return browser.wait(() => browser.isElementPresent(by.id('reports-tab')), 10000);
+  };
+
+  describe('submits new sms messages', () => {
+
+    beforeEach(done => {
+      browser.ignoreSynchronization = true;
+      const body = {
+        messages: [{
+          from: PHONE,
+          content: 'R Siobhan 12',
+          id: 'a'
+        }]
+      };
+      utils.updateSettings(CONFIG)
+        .then(() => protractor.promise.all(DOCS.map(utils.saveDoc)))
+        .then(results => results.forEach(result => savedUuids.push(result.id)))
+        .then(() => submit(body))
+        .then(() => refresh)
+        .then(done)
+        .catch(done);
+    });
+
+    afterEach(done => {
+      utils.revertSettings()
+        .then(() => protractor.promise.all(savedUuids.map(utils.deleteDoc)))
+        .then(refresh)
+        .then(done)
+        .catch(done);
+    });
+
+    const checkItemSummary = () => {
+      const summaryElement = element(by.css('#reports-content .item-summary'));
+      expect(summaryElement.element(by.css('.name')).getText()).toBe(CAROL.name);
+      expect(summaryElement.element(by.css('.phone')).getText()).toBe(CAROL.phone);
+      expect(summaryElement.element(by.css('.position a')).getText()).toBe(BOB_PLACE.name);
+      expect(summaryElement.element(by.css('.detail > div:last-child')).getText()).toBe(FORM_NAME);
+      expect(summaryElement.element(by.css('.detail .status .fa-circle.green-dot')).isDisplayed()).toBeTruthy();
+    };
+
+    const checkAutoResponse = () => {
+      const taskElement = element(by.css('#reports-content .details > ul'));
+      expect(taskElement.element(by.css('.task-list > li > ul > li')).getText()).toBe('Thank you for registering Siobhan');
+      expect(taskElement.element(by.css('.task-list .task-state .state.pending')).isDisplayed()).toBeTruthy();
+      expect(taskElement.element(by.css('.task-list .task-state .recipient')).getText()).toBe(' to +64271234567');
+    };
+
+    const checkScheduledTask = (childIndex, title, message) => {
+      const taskElement = element(by.css('#reports-content .details .scheduled-tasks > ul > li:nth-child(' + childIndex + ')'));
+      expect(taskElement.element(by.css('p')).getText()).toContain(title);
+      expect(taskElement.element(by.css('.task-list li > ul > li')).getText()).toBe(message);
+      expect(taskElement.element(by.css('.task-list li .task-state .state.scheduled')).isDisplayed()).toBeTruthy();
+      expect(taskElement.element(by.css('.task-list li .task-state .recipient')).getText()).toBe(' to +64271234567');
+    };
+
+    it('shows content', () => {
+      browser.driver.navigate().refresh();
+      browser.wait(() => browser.isElementPresent(by.id('reports-tab')), 10000);
+      element(by.id('reports-tab')).click();
+
+      // refresh - live list only updates on changes but changes are disabled for e2e
+      browser.driver.navigate().refresh();
```

Comment:
Again. Gave up. Got green. Committed. One day I'll feel confident enough to answer that question and I can clean up all our tests.